### PR TITLE
Fixed detection of B8G8R8 based formats.

### PR DESCRIPTION
**Pull request by ptasev**
- Fixed detection of B8GR8 formats and R8.
- Added BGRA8 decoder/encoder and wired it to DDS
- Fixed issue where DDS mip width/height could become 0

I'll need help to wire this into KTX since I'm not familiar with the format.

### CODE REVIEW

**Comment by Nominom** [comment]
Thanks for fixing this, that was a silly mistake. Also nice to have BGRA support. Would you like to write tests for dds BGRA encoding/decoding? I can also do it. I'll wire it up to ktx after I merge this.

**Comment by ptasev** [comment]
Sounds good! I'm not planning on adding the tests.

**Comment by Nominom** [comment]
Alright, I'll merge this and add tests + ktx.

File: BCnEnc.Net/Decoder/RawDecoder.cs [hunk 1]
```diff
@@ -177,4 +177,41 @@ public Rgba32[] Decode(ReadOnlyMemory<byte> data, int pixelWidth, int pixelHeigh
...
+		public Rgba32[] Decode(ReadOnlyMemory<byte> data, int pixelWidth, int pixelHeight, OperationContext context)
+		{
+			var output = new Rgba32[pixelWidth * pixelHeight];
+
+			// HINT: Ignoring parallel execution since we wouldn't gain performance from it.
+
+			var span = data.Span;
+			for (var i = 0; i < output.Length; i++)
+			{
+				if (context.CancellationToken.IsCancellationRequested)
```
Comment (onepiecefreak3) [inline BCnEnc.Net/Decoder/RawDecoder.cs:203]: Please use ``context.Cancellation.ThrowIfCancellationRequested()`` instead.

File: BCnEnc.Net/Encoder/RawEncoders.cs [hunk 2]
```diff
@@ -139,4 +139,36 @@ public uint GetGlTypeSize()
...
+			{
+				output[i * 4] = pixels[i].B;
+				output[i * 4 + 1] = pixels[i].G;
+				output[i * 4 + 2] = pixels[i].R;
+				output[i * 4 + 3] = pixels[i].A;
+			}
+			return output;
+		}
+
+		public GlInternalFormat GetInternalFormat()
```
Comment (onepiecefreak3) [inline BCnEnc.Net/Encoder/RawEncoders.cs:158]: Please use the same style as in the other encoders by using brackets and ``return``. This makes it look more consistent.

**Comment by onepiecefreak3** [comment]
Thanks, that resolved everything I noticed.

**Comment by Nominom** [comment]
Nice, I'll merge this now.

**Pull request merged**